aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -179,15 +179,25 @@ impl<'r, 'l> SessionExt<'r, 'l> {
                                   layout: MoveTypeLayout,
                                   has_aggregator_lifting: bool|
          -> PartialVMResult<BytesWithResourceLayout> {
-            value
-                .simple_serialize(&layout)
-                .map(Into::into)
-                .map(|bytes| (bytes, has_aggregator_lifting.then_some(Arc::new(layout))))
+            let serialization_result = if has_aggregator_lifting {
+                // We allow serialization of native values here because we want to
+                // temporarily store native values (via encoding to ensure deterministic
+                // gas charging) in block storage.
+                serialize_and_allow_delayed_values(&value, &layout)
+                    .map(|bytes| (bytes, Some(Arc::new(layout))))
+            } else {
+                // Otherwise, there should ne no native values so ensure
+                // serialization fails here.
+                value.simple_serialize(&layout).map(|bytes| (bytes, None))
+            };
+            serialization_result
+                .map(|(bytes, maybe_layout)| (bytes.into(), maybe_layout))
```

Comment:
nit: we can just have bytes.into() twice above instead of (bytes, ..) and get rid of this line?